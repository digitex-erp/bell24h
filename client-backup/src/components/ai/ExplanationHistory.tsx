import React, { useState, useEffect, useRef } from 'react';
import {
  Box,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Paper,
  Typography,
  IconButton,
  TablePagination,
  TableSortLabel,
  CircularProgress,
  useTheme,
  useMediaQuery
} from '@mui/material';
import { Refresh as RefreshIcon, Visibility as VisibilityIcon } from '@mui/icons-material';
import { format } from 'date-fns';

interface Explanation {
  id: string;
  timestamp: string;
  model: string;
  confidence: number;
  summary: string;
  features: Record<string, number>;
}

interface ExplanationHistoryProps {
  onViewExplanation: (explanation: Explanation) => void;
  refreshInterval?: number;
}

export const ExplanationHistory: React.FC<ExplanationHistoryProps> = ({
  onViewExplanation,
  refreshInterval = 30000
}) => {
  const [explanations, setExplanations] = useState<Explanation[]>([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(10);
  const [orderBy, setOrderBy] = useState<keyof Explanation>('timestamp');
  const [order, setOrder] = useState<'asc' | 'desc'>('desc');

  // Theme and responsive hooks
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
  const loadingRef = useRef<HTMLDivElement>(null);
  const tableRef = useRef<HTMLDivElement>(null);

  const fetchExplanations = async () => {
    try {
      setLoading(true);
      setError(null);
      
      const queryParams = new URLSearchParams({
        page: (page + 1).toString(),
        limit: rowsPerPage.toString(),
        sortBy: orderBy,
        sortOrder: order
      });

      const response = await fetch(`/api/explanations?${queryParams.toString()}`);
      
      if (!response.ok) {
        throw new Error('Failed to fetch explanations');
      }

      const data = await response.json();
      setExplanations(data.items);
    } catch (err) {
      setError(err instanceof Error ? err.message : 'An error occurred');
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    fetchExplanations();
    const interval = setInterval(fetchExplanations, refreshInterval);
    return () => clearInterval(interval);
  }, [page, rowsPerPage, orderBy, order, refreshInterval]);

  const handleChangePage = (event: unknown, newPage: number) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  const handleSort = (property: keyof Explanation) => {
    const isAsc = orderBy === property && order === 'asc';
    setOrder(isAsc ? 'desc' : 'asc');
    setOrderBy(property);
  };

  const handleRefresh = () => {
    fetchExplanations();
  };

  if (error) {
    return (
      <Box sx={{ p: 2 }}>
        <Typography color="error">{error}</Typography>
        <IconButton onClick={handleRefresh}>
          <RefreshIcon />
        </IconButton>
      </Box>
    );
  }

  return (
    <Box sx={{ width: '100%' }}>
      <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center', mb: 2 }}>
        <Typography variant="h6">Explanation History</Typography>
        <IconButton onClick={handleRefresh} disabled={loading}>
          <RefreshIcon />
        </IconButton>
      </Box>

      <TableContainer component={Paper} ref={tableRef}>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>
                <TableSortLabel
                  active={orderBy === 'timestamp'}
                  direction={orderBy === 'timestamp' ? order : 'asc'}
                  onClick={() => handleSort('timestamp')}
                >
                  Timestamp
                </TableSortLabel>
              </TableCell>
              <TableCell>
                <TableSortLabel
                  active={orderBy === 'model'}
                  direction={orderBy === 'model' ? order : 'asc'}
                  onClick={() => handleSort('model')}
                >
                  Model
                </TableSortLabel>
              </TableCell>
              <TableCell>
                <TableSortLabel
                  active={orderBy === 'confidence'}
                  direction={orderBy === 'confidence' ? order : 'asc'}
                  onClick={() => handleSort('confidence')}
                >
                  Confidence
                </TableSortLabel>
              </TableCell>
              <TableCell>Summary</TableCell>
              <TableCell>Actions</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {loading ? (
              <TableRow>
                <TableCell colSpan={5} align="center">
                  <CircularProgress />
                </TableCell>
              </TableRow>
            ) : explanations.length === 0 ? (
              <TableRow>
                <TableCell colSpan={5} align="center">
                  No explanations available
                </TableCell>
              </TableRow>
            ) : (
              explanations.map((explanation) => (
                <TableRow key={explanation.id}>
                  <TableCell>
                    {format(new Date(explanation.timestamp), 'MMM d, yyyy HH:mm')}
                  </TableCell>
                  <TableCell>{explanation.model}</TableCell>
                  <TableCell>{(explanation.confidence * 100).toFixed(1)}%</TableCell>
                  <TableCell>{explanation.summary}</TableCell>
                  <TableCell>
                    <IconButton
                      size="small"
                      onClick={() => onViewExplanation(explanation)}
                    >
                      <VisibilityIcon />
                    </IconButton>
                  </TableCell>
                </TableRow>
              ))
            )}
          </TableBody>
        </Table>
      </TableContainer>

      <TablePagination
        rowsPerPageOptions={[5, 10, 25]}
        component="div"
        count={-1}
        rowsPerPage={rowsPerPage}
        page={page}
        onPageChange={handleChangePage}
        onRowsPerPageChange={handleChangeRowsPerPage}
      />
    </Box>
  );
};