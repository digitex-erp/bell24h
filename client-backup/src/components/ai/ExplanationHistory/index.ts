export { default } from './ExplanationHistory';
export * from './ExplanationHistory';
