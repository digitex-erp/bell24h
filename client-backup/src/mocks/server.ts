import { setupServer } from 'msw/node';
import { http, HttpResponse } from 'msw';
import { categories } from '../config/categories';

// Mock API endpoints
export const handlers = [
  // Auth endpoints
  http.post('/api/auth/login', () => {
    return HttpResponse.json({
      token: 'mock-token',
      user: { id: 1, name: 'Test User' }
    });
  }),

  // RFQ endpoints
  http.get('/api/rfq', ({ request }) => {
    const url = new URL(request.url);
    const category = url.searchParams.get('category');
    
    if (category) {
      const categoryConfig = categories.find(c => c.id === category);
      return HttpResponse.json(categoryConfig?.mockupRFQs || []);
    }
    
    return HttpResponse.json(categories.flatMap(c => c.mockupRFQs));
  }),

  http.get('/api/rfq/:id', ({ params }) => {
    const rfq = categories
      .flatMap(c => c.mockupRFQs)
      .find(r => r.id === params.id);
    
    if (!rfq) {
      return new HttpResponse(null, { status: 404 });
    }
    
    return HttpResponse.json(rfq);
  }),

  http.post('/api/rfq', async ({ request }) => {
    const body = await request.json();
    return HttpResponse.json({
      ...body,
      id: Math.random().toString(36).substr(2, 9),
      createdAt: new Date().toISOString(),
      updatedAt: new Date().toISOString(),
      status: 'open'
    });
  }),

  http.post('/api/rfq/:id/respond', async ({ request }) => {
    const body = await request.json();
    return HttpResponse.json({
      ...body,
      id: Math.random().toString(36).substr(2, 9),
      createdAt: new Date().toISOString(),
      status: 'pending'
    });
  }),

  // Quote endpoints
  http.get('/api/quotes', () => {
    return HttpResponse.json({
      items: [
        {
          id: '1',
          text: 'Test quote 1',
          author: 'Author 1',
          category: 'test'
        },
        {
          id: '2',
          text: 'Test quote 2',
          author: 'Author 2',
          category: 'test'
        }
      ],
      total: 2,
      page: 1,
      pageSize: 10
    });
  }),

  http.get('/api/quotes/:id', ({ params }) => {
    return HttpResponse.json({
      id: params.id,
      text: 'Test quote',
      author: 'Test Author',
      category: 'test'
    });
  }),

  // AI endpoints
  http.post('/api/ai/analyze', () => {
    return HttpResponse.json({
      decision: 'approve',
      confidence: 0.85,
      reasoning: 'Test reasoning',
      timestamp: new Date().toISOString()
    });
  }),

  // Explainability endpoints
  http.get('/api/explainability', () => {
    return HttpResponse.json({
      items: [
        {
          id: '1',
          timestamp: '2024-01-01T00:00:00Z',
          model: 'test-model',
          confidence: 0.95,
          summary: 'Test explanation',
          features: {
            'feature1': 0.8,
            'feature2': 0.6
          }
        }
      ],
      total: 1,
      page: 1,
      pageSize: 10
    });
  }),

  // Feedback endpoints
  http.post('/api/feedback', () => {
    return HttpResponse.json({
      success: true,
      message: 'Feedback submitted successfully'
    });
  }),

  http.get('/api/feedback', () => {
    return HttpResponse.json({
      items: [
        {
          id: '1',
          timestamp: '2024-01-01T00:00:00Z',
          rating: 5,
          comment: 'Great service!',
          category: 'general'
        }
      ],
      total: 1,
      page: 1,
      pageSize: 10
    });
  }),

  // Analytics endpoints
  http.get('/api/analytics/dashboard', () => {
    return HttpResponse.json({
      totalTransactions: 150,
      activeSuppliers: 45,
      pendingRFQs: 12,
      revenue: 250000,
    });
  }),

  // Supplier endpoints
  http.get('/api/suppliers/risk', () => {
    return HttpResponse.json({
      suppliers: [
        {
          id: 1,
          name: 'Supplier A',
          riskScore: 0.85,
          status: 'active',
        },
        {
          id: 2,
          name: 'Supplier B',
          riskScore: 0.65,
          status: 'active',
        },
      ],
    });
  }),

  // Transaction endpoints
  http.get('/api/transactions', () => {
    return HttpResponse.json({
      transactions: [
        {
          id: 1,
          amount: 5000,
          status: 'completed',
          date: '2024-03-20',
        },
        {
          id: 2,
          amount: 3000,
          status: 'pending',
          date: '2024-03-21',
        },
      ],
    });
  }),

  // Performance metrics endpoints
  http.get('/api/performance', () => {
    return HttpResponse.json({
      metrics: {
        responseTime: 250,
        uptime: 99.9,
        errorRate: 0.1,
        userSatisfaction: 4.5,
      },
    });
  }),
];

export const server = setupServer(...handlers);
