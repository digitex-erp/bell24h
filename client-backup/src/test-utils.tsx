import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Mock ChakraProvider component
const ChakraProvider = ({ children }: { children: React.ReactNode }) => {
  return <div data-testid="chakra-provider">{children}</div>;
};

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <ChakraProvider>
        <Router>
          {children}
        </Router>
      </ChakraProvider>
    </QueryClientProvider>
  );
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
