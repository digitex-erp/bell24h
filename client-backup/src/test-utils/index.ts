// Re-export all test utilities
export * from './custom-render';
export * from './component-utils';
export * from './user-event-utils';
export * from './mock-data';
// Add other test utility exports here as needed
