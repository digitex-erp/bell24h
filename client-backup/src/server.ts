console.log('=== Bell24H.com server.ts is running and starting up... ===');

process.on('uncaughtException', (err) => {
  console.error('Uncaught Exception:', err);
});
process.on('unhandledRejection', (reason) => {
  console.error('Unhandled Rejection:', reason);
});

import 'dotenv/config'; // Added for environment variables
import express from 'express';
import { createServer } from 'http';
import { Server } from 'socket.io';
import dotenv from 'dotenv';
import path from 'path';
import { fileURLToPath } from 'url';
import helmet from 'helmet';
import cors from 'cors';
import compression from 'compression';

import { validate } from './middleware/validateRequest'; // <-- Import middleware
import { createRfqSchema } from './api/rfqs/rfq.schema';        // <-- Import schema
import { registerRoutes } from './server/routes'; // Correct import for route registration
import { connectDatabase } from './database.ts'; // Assuming database connection is here
import { N8nApp } from './n8n-app.ts'; // Assuming n8n app definition is here

dotenv.config();

const app = express();
const httpServer = createServer(app);
const io = new Server(httpServer, {
  cors: {
    origin: '*',
  },
});

// Basic Middleware
app.use(helmet());
app.use(cors({
  origin: process.env.CORS_ORIGIN || 'http://localhost:3000',
  credentials: true
}));
app.use(compression());
app.use(express.json());
app.use(express.urlencoded({ extended: true }));

// Serve static files from the public directory
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const publicPath = path.join(__dirname, '../public');
console.log('Serving static files from', publicPath);
app.use(express.static(publicPath));

// API routes will be registered by registerRoutes function after server starts

// Register RFQ webhook endpoint that can trigger n8n workflows
app.post(
  '/api/rfq/new',
  validate(createRfqSchema) as express.RequestHandler,
  async (req: express.Request, res: express.Response) => {
  const { rfqData } = req.body as unknown as { rfqData: any };

  try {
    // This would normally be handled by a n8n workflow
    // But here we're showing what would happen in the workflow
    // In a real implementation, you'd make a request to trigger the workflow
      const prisma = (req.app as any).locals.prisma;
    await prisma?.rfq.create({ data: rfqData });

    // Simulate sending an email
    console.log(`Would send email to ${rfqData.buyerEmail}`);

      // Broadcast via WebSocket
      io.emit('rfq-submitted', rfqData);

      res.status(200).json({ status: 'success', message: 'RFQ submitted and workflow triggered' });
    } catch (error) {
      console.error('Error processing RFQ:', error);
      res.status(500).json({ status: 'error', message: 'Failed to process RFQ' });
    }
  });

  // Register the root route
  console.log('Registering root route...');
  app.get('/', (_req, res) => {
    res.sendFile(path.join(publicPath, 'index.html'));
  });

  // Connect to database
  connectDatabase();

// Initialize n8n (Option B: Integrated approach)
const initN8n = async () => {
    try {
      const n8n = new N8nApp({
        express: app,
        endpoint: '/api/workflow',
        credentialsOverwrite: {
          basicAuth: {
            active: process.env.N8N_BASIC_AUTH_ACTIVE === 'true',
            user: process.env.N8N_BASIC_AUTH_USER || 'admin',
            password: process.env.N8N_BASIC_AUTH_PASSWORD || 'your-secure-password'
          }
        }
      });

      await n8n.init();
      console.log('✅ n8n workflow engine initialized successfully');
      console.log('n8n Username:', process.env.N8N_USERNAME);
      console.log('n8n Port:', process.env.N8N_PORT);
    } catch (error) {
      console.error('❌ Failed to initialize n8n:', error);
    }
  };

  // Start the server
  const PORT = process.env.PORT || 3000;
  console.log('About to call httpServer.listen...');
  httpServer.listen(PORT, async () => {
    console.log('============================================================');
    console.log(`🚀 SERVER SUCCESSFULLY STARTED on port ${PORT}`);
    console.log(`🔗 Access: http://localhost:${PORT}`);
    console.log(`🔗 n8n Dashboard: http://localhost:${PORT}/api/workflow`);
    console.log('============================================================');

    // Initialize n8n after server starts
    await initN8n();
    // Register API routes after server and n8n start
    await registerRoutes(app);
  });

  // WebSocket connection handler
  io.on('connection', (socket) => {
    console.log('New WebSocket connection established');

    socket.on('join-room', (roomId) => {
      socket.join(roomId);
      console.log(`Socket joined room: ${roomId}`);
    });

    socket.on('leave-room', (roomId) => {
      socket.leave(roomId);
      console.log(`Socket left room: ${roomId}`);
    });

    socket.on('disconnect', () => {
      console.log('WebSocket connection closed');
    });
  });

  // Helper function to broadcast WebSocket messages (used by n8n workflows)
  export const broadcastWebSocketMessage = (event: string, data: any) => {
    io.emit(event, data);
    console.log(`Broadcasted WebSocket message: ${event}`);
  };

  export { io }; // Export io for use in other modules