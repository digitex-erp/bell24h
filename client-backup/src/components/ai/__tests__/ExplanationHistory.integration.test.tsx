import { render, screen, fireEvent } from '../../../test-utils';
import { ExplanationHistory } from '../ExplanationHistory';
import { server } from '../../../mocks/server';
import { http, HttpResponse } from 'msw';

// Setup MSW for API mocking
const mockExplanations = [
  {
    id: '1',
    modelType: 'Risk Assessment',
    timestamp: '2023-06-15T10:30:00Z',
    summary: 'Low risk profile detected',
  },
  {
    id: '2',
    modelType: 'Fraud Detection',
    timestamp: '2023-06-15T11:45:00Z',
    summary: 'Potential fraud pattern identified',
  },
];

describe('ExplanationHistory - Integration Tests', () => {
  beforeAll(() => {
    // Enable API mocking before all tests
    server.use(
      http.get('/api/explanations', () => {
        return HttpResponse.json({
          items: mockExplanations,
          totalItems: mockExplanations.length,
        });
      })
    );
  });

  afterEach(() => {
    // Reset any runtime request handlers we may add during the tests
    server.resetHandlers();
  });

  it('should fetch and display explanations', async () => {
    render(<ExplanationHistory />);
    
    // Check loading state
    expect(screen.getByRole('progressbar')).toBeInTheDocument();
    
    // Wait for data to load
    const firstItem = await screen.findByText(mockExplanations[0].modelType);
    expect(firstItem).toBeInTheDocument();
    
    // Verify all items are rendered
    mockExplanations.forEach(item => {
      expect(screen.getByText(item.modelType)).toBeInTheDocument();
      expect(screen.getByText(item.summary)).toBeInTheDocument();
    });
    
    // Verify pagination shows correct count
    expect(screen.getByText(/1–2 of 2/)).toBeInTheDocument();
  });

  it('should handle API errors gracefully', async () => {
    // Mock error response
    server.use(
      http.get('/api/explanations', () => {
        return HttpResponse.json(
          { message: 'Internal server error' },
          { status: 500 }
        );
      })
    );

    render(<ExplanationHistory />);
    
    // Verify error message is shown - the component shows the actual error message from the API
    const errorMessage = await screen.findByText(/API request failed with status 500/);
    expect(errorMessage).toBeInTheDocument();
  });

  it('should support sorting', async () => {
    // Mock the API response with sorting support
    server.use(
      http.get('/api/explanations', ({ request }) => {
        const url = new URL(request.url);
        const sortBy = url.searchParams.get('sortBy') || 'timestamp';
        const sortOrder = url.searchParams.get('sortOrder') || 'desc';
        
        // Create a sorted copy of the mock data based on the sort parameters
        const sortedItems = [...mockExplanations].sort((a, b) => {
          if (sortBy === 'modelType') {
            return sortOrder === 'asc' 
              ? a.modelType.localeCompare(b.modelType)
              : b.modelType.localeCompare(a.modelType);
          } else if (sortBy === 'timestamp') {
            return sortOrder === 'asc'
              ? new Date(a.timestamp).getTime() - new Date(b.timestamp).getTime()
              : new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime();
          }
          return 0;
        });
        
        return HttpResponse.json({
          items: sortedItems,
          totalItems: sortedItems.length,
        });
      })
    );
    
    // Render the component
    render(<ExplanationHistory />);
    
    // Wait for initial data to load
    await screen.findByText(mockExplanations[0].modelType);
    
    // Get the model type header and click it to sort by model type
    const modelTypeHeader = screen.getByText('Model Type').closest('th');
    if (!modelTypeHeader) {
      throw new Error('Could not find Model Type header');
    }
    
    // Click to sort by model type (first click sorts ascending)
    fireEvent.click(modelTypeHeader);
    
    // Wait for the sorted data to be displayed
    await screen.findByText(mockExplanations[1].modelType);
    
    // Get all table rows (skip header row)
    const rows = screen.getAllByRole('row').slice(1);
    expect(rows).toHaveLength(2);
    
    // Get the model type cells from each row (they should be the second cell in each row)
    const modelTypeCells = rows.map(row => 
      row.querySelector('td:nth-child(2)')
    );
    
    // Verify the cells contain the expected model types in the correct order
    // After first click, it should be sorted by modelType in ascending order
    const expectedOrder = [...mockExplanations]
      .sort((a, b) => a.modelType.localeCompare(b.modelType))
      .map(item => item.modelType);
    
    expect(modelTypeCells[0]).toHaveTextContent(expectedOrder[0]);
    expect(modelTypeCells[1]).toHaveTextContent(expectedOrder[1]);
  });

  it('should support pagination', async () => {
    // Mock handler that returns all items on the first page since the default page size is 10
    const requestHandler = http.get('/api/explanations', ({ request }) => {
      const url = new URL(request.url);
      const page = parseInt(url.searchParams.get('page') || '1');
      const limit = parseInt(url.searchParams.get('limit') || '10');
      
      console.log(`API Request - page: ${page}, limit: ${limit}`);
      
      // For testing, we'll return both items on the first page since the default limit is 10
      return HttpResponse.json({
        items: [mockExplanations[0], mockExplanations[1]],
        totalItems: 2,
      });
    });
    
    server.use(requestHandler);

    // Render the component
    render(<ExplanationHistory />);
    
    // Wait for items to load
    const firstItem = await screen.findByText(mockExplanations[0].modelType);
    expect(firstItem).toBeInTheDocument();
    
    // Since both items fit on one page (limit=10), we should see both items
    const secondItem = await screen.findByText(mockExplanations[1].modelType);
    expect(secondItem).toBeInTheDocument();
    
    // Check pagination shows 1-2 of 2 since both items are on the first page
    const paginationText = screen.getByText(/1–2 of 2/);
    expect(paginationText).toBeInTheDocument();
    
    // Get pagination controls
    const nextButton = screen.getByRole('button', { name: /next page/i }) as HTMLButtonElement;
    const prevButton = screen.getByRole('button', { name: /previous page/i }) as HTMLButtonElement;
    
    // The next button should be disabled since we're showing all items on one page
    expect(nextButton).toBeDisabled();
    // The previous button should be disabled since we're on the first page
    expect(prevButton).toBeDisabled();
  });

  it('matches snapshot', async () => {
    // Mock the API response
    server.use(
      http.get('/api/explanations', () => {
        return HttpResponse.json({
          items: mockExplanations,
          totalItems: mockExplanations.length,
        });
      })
    );

    // Render the component
    const { container } = render(<ExplanationHistory />);
    
    // Wait for data to load
    await screen.findByText(mockExplanations[0].modelType);
    
    // Take a snapshot of the rendered component
    expect(container).toMatchSnapshot();
  });
});
