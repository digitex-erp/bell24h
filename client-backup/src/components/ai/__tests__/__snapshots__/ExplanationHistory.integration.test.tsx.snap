// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExplanationHistory - Integration Tests > matches snapshot 1`] = `
<div>
  <div
    data-testid="chakra-provider"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-15htl6s-MuiPaper-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Explanation History
      </h5>
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          aria-label="explanation history table"
          class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  ID
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  Model Type
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort="descending"
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  Timestamp
                  <span
                    class="MuiBox-root css-3hetpk"
                  >
                    sorted descending
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc css-3l415a-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  Summary
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Risk Assessment
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                15/6/2023, 4:00:00 pm
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Low risk profile detected
              </td>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Fraud Detection
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                15/6/2023, 5:15:00 pm
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Potential fraud pattern identified
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
            id=":ra:"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          >
            <div
              aria-controls=":rb:"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby=":ra: :r9:"
              class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id=":r9:"
              role="combobox"
              tabindex="0"
            >
              10
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="10"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            1â€“2 of 2
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
